tests/Incoming/Test/Structure/StructureFactoryTest.php

```diff
@@ -122,4 +122,17 @@ public function testBuildWithInvalidStructuralType()
 
         $structure = (new StructureFactory)->build($data);
     }
+    
+    /**
+     * @see https://github.com/Rican7/incoming/pull/3
+     */
+    public function testBuildWithTraversableAndMixedKeys()
+    {
+        $data = ['name' => 'markus', 0 => 'a crazy mixed in key'];
+
+        $structure = (new StructureFactory)->build(new ArrayIterator($data));
+
+        $this->assertInstanceOf('Incoming\Structure\Map', $structure);
```

Comment:
I'm not a huge fan of `assertInstanceOf` due to the way that it forces fully qualified class names. I use it for PHP 5.6 projects where I can use `ClassName::class`, but I opt to stay away from it otherwise.